applications/StructuralMechanicsApplication/python_scripts/structural_response.py

```diff
@@ -75,6 +75,12 @@ def __init__(self, identifier, response_settings, model_part):
         self.primal_analysis = structural_mechanics_analysis.StructuralMechanicsAnalysis(model, ProjectParametersPrimal)
         self.primal_model_part.AddNodalSolutionStepVariable(SHAPE_SENSITIVITY)
 
+        self._Check()
```

Comment:
I find it very hard to read and understand why the name of the self.primal_model_part and the self.primal_analysis.... should not be be same. 

We could express a bit clearer I think. The problem is that between the two inputs "model_part" and "response_settings" there might be an inconsistency w.r.t. the model part name. The check should reflect  this problem and compare directly the input (model_part.Name and ProjectParametersPrimal [...]["model_part"]) rather than the already attributed values (self.primal_model_part, self.primal_analysis). On top of that I would give a more exressive name to the function - like CheckInputSettingsForConsistency(). While I understand that Check more looks like a general standard function as often used in Kratos, it is in fact doing something very special here. So no need to imply generality here, I guess.

Reply:
@armingeiser such modifications are unrelated to this PR and should be discussed in a separate PR